### ISSUE TITLE
onLongPress are not working properly in nested Text components

### ISSUE THREAD

**Issue by Dolavs** [facebook/react-native#11839]
### What I did:
- Install AwesomeProject
- removed the default text 
- make a new Text component inside another Text component as follow: 
```
export default class AwesomeProject extends Component {
  render() {
    return (
      <View style={styles.container}>
        <Text style={styles.instructions}>
          <Text style={styles.welcome} onPress={() => Alert.alert('It\'s a short Press, not long one')} onLongPress={() => Alert.alert('Long Press')}>
            Click Here
          </Text>
        </Text>
      </View>
    );
  }
}
```
- Run the app in iPhone 7 (must be with **3DTouch**)
- Long press on 'Click Here' 
### Expected result:
Open an alert.
### Actual result:
Do nothing
## Reproduction
The bug aren't reproduce in simulator, only in real device with 3DTouch
[Sample App](https://rnplay.org/apps/eISvUw)
## Additional Information

* React Native version: 0.38+
* Platform: iOS (3DTouch supported devices)

**Comment by TGPSKI** [comment]
Bump. Having the same issue and its causing a lot of pain for me. I've spent a few days now trying to fix this and I don't have a solution. 

RN 0.40, iPhone 7

**Comment by TGPSKI** [comment]
https://github.com/facebook/react-native/issues/11462

**Comment by TGPSKI** [comment]
@evollu has a fix with https://github.com/facebook/react-native/pull/11530. I made these changes locally and nested text onPress and onLongPress fires correctly. 

Still waiting on this merge into master. I'd say this is a pretty critical bug for anyone wanting to use links in nested text, which should be a lot of people.

**Comment by hramos** [comment]
Hi there! This issue is being closed because it has been inactive for a while.

But don't worry, it will live on with Canny! Check out its new home: https://react-native.canny.io/feature-requests/p/onlongpress-are-not-working-properly-in-nested-text-components
